lib/components/user/monitored-trip/trip-status.js

```diff
@@ -20,9 +21,25 @@ const StyledFooterButton = styled(Button)`
 `
 
 export function FooterButton({ bsStyle, glyphIcon, onClick, text }) {
+  const [loading, setLoading] = useState(false)
+  const clickHandler = () => {
+    setLoading(true)
+    onClick()
+  }
+
+  useEffect(() => {
+    setLoading(false)
+  }, [text])
```

Comment:
This may work now, however I'd rather set the state based on actual completion of the underlying action.